test/onnx/onnx_import_quant.in.cpp

```diff
@@ -338,3 +338,102 @@ NGRAPH_TEST(onnx_${BACKEND_NAME}, model_qlinear_matmul_3d)
+    auto test_case = ngraph::test::NgraphTestCase(function, "${BACKEND_NAME}");
+
+    test_case.add_input(std::vector<uint8_t>{11, 7, 3, 10, 6, 2, 9, 5, 1, 8, 4, 0}); // a
+    test_case.add_input(std::vector<uint8_t>{1, 4, 2, 5, 3, 6});                     // b
+    test_case.add_input(std::vector<uint8_t>{12});                                   // a_zero_point
+    test_case.add_input(std::vector<uint8_t>{0});                                    // b_zero_point
+
+    test_case.add_expected_output(
+        {4, 2}, std::vector<int32_t>{-38, -83, -44, -98, -50, -113, -56, -128}); // y
+    test_case.run();
+}
+
+NGRAPH_TEST(onnx_${BACKEND_NAME}, model_matmul_integer_zero_point_zero)
+{
+    auto function = onnx_import::import_onnx_model(
+        file_util::path_join(SERIALIZED_ZOO, "onnx/matmul_integer.prototxt"));
+    auto test_case = ngraph::test::NgraphTestCase(function, "${BACKEND_NAME}");
+
+    test_case.add_input(std::vector<uint8_t>{11, 7, 3, 10, 6, 2, 9, 5, 1, 8, 4, 0}); // a
+    test_case.add_input(std::vector<uint8_t>{1, 4, 2, 5, 3, 6});                     // b
+    test_case.add_input(std::vector<uint8_t>{0});                                    // a_zero_point
+    test_case.add_input(std::vector<uint8_t>{0});                                    // b_zero_point
+
+    test_case.add_expected_output({4, 2},
+                                  std::vector<int32_t>{34, 97, 28, 82, 22, 67, 16, 52}); // y
+    test_case.run();
+}
+
+NGRAPH_TEST(onnx_${BACKEND_NAME}, model_matmul_integer_no_zero_point)
+{
+    auto function = onnx_import::import_onnx_model(
+        file_util::path_join(SERIALIZED_ZOO, "onnx/matmul_integer_no_zero_point.prototxt"));
+    auto test_case = ngraph::test::NgraphTestCase(function, "${BACKEND_NAME}");
+
+    test_case.add_input(std::vector<uint8_t>{11, 7, 3, 10, 6, 2, 9, 5, 1, 8, 4, 0}); // a
+    test_case.add_input(std::vector<uint8_t>{1, 4, 2, 5, 3, 6});                     // b
+
+    test_case.add_expected_output({4, 2},
+                                  std::vector<int32_t>{34, 97, 28, 82, 22, 67, 16, 52}); // y
+    test_case.run();
+}
+
+NGRAPH_TEST(onnx_${BACKEND_NAME}, model_matmul_integer_scalar)
+{
+    auto function = onnx_import::import_onnx_model(
+        file_util::path_join(SERIALIZED_ZOO, "onnx/matmul_integer_scalar.prototxt"));
+    auto test_case = ngraph::test::NgraphTestCase(function, "${BACKEND_NAME}");
+
+    test_case.add_input(std::vector<uint8_t>{11}); // a
+    test_case.add_input(std::vector<uint8_t>{13}); // b
+    test_case.add_input(std::vector<uint8_t>{12}); // a_zero_point
+    test_case.add_input(std::vector<uint8_t>{12}); // b_zero_point
+
+    test_case.add_expected_output({1, 1}, std::vector<int32_t>{-1}); // y
+    test_case.run();
+}
+
+NGRAPH_TEST(onnx_${BACKEND_NAME}, model_matmul_integer_4d)
+{
+    auto function = onnx_import::import_onnx_model(
+        file_util::path_join(SERIALIZED_ZOO, "onnx/matmul_integer_4d.prototxt"));
```

Comment:
Please use input tensors with somewhat more complicated shape. This will let us to catch errors where we are messing multiplied axes.

Reply:
What shapes would you suggest? Can you give an example?